cmake/ElektraCache.cmake

```diff
@@ -103,6 +103,14 @@ if (FINDEX GREATER -1 OR BINDINGS MATCHES "ALL")
 	set (BINDINGS_FORCE FORCE)
 endif()
 
+list (FIND BINDINGS "IO" FINDEX)
```

Comment:
Ok. I'll look into that.

Reply:
Alright. Seems like I'll have to update all bindings then. There is too much interference when I try to only change the behavior for the I/O bindings. I'll commit my changes and then change this.

Would it be nice to have build categories like plugins do (e.g. ALL;-EXPERIMENTAL)?

Reply:
> I'll commit my changes and then change this.

Thank you, such infrastructure work is really important.

> Would it be nice to have build categories like plugins do (e.g. ALL;-EXPERIMENTAL)?

Yes, categories would be important, especially categories like IO, SWIG, etc. to group bindings based on same technologies.

Ideally there would be reuse with the add_plugin logic so that we can easily extend to add_tool without duplicating CMake code three times.

Reply:
Ok. I will try to refactor `plugin_check_if_included` to be more generic. I think it would be nice to keep the change as minimal as possible. For example:
```cmake
list (FIND BINDINGS "io_uv" FINDEX)
if (FINDEX GREATER -1)
	add_subdirectory (uv)
endif ()
```
to 
```cmake
set (INCLUDED "")
generic_check_if_included (BINDINGS "io_uv") # with additional parameters
if (INCLUDED)
	add_subdirectory (uv)
endif ()
```

Reply:
Yes, this is certainly an improvement to status quo. Please make sure to not directly modify BINDINGS (because this is a user-specified CACHE variable we want to preserve exactly in the way like the user wrote it.) So running cmake again should find more bindings if dependencies were installed meanwhile.

Some remarks about your API "generic_check_if_included":

- In CMake you usually pass the OUTVARIABLE so that one can choose where the information is written to (and not use a hardcoded string like "INCLUDED".) plugin_check_if_included has the same problem, though.
- `generic_` can be omitted.
- It would be okay if "bindings" is a part of the function name. It is enough to reuse the implementation (so that plugin_check_if_included, binding_check_if_included, and tool_check_if_included is only very short calling the generic functions)  -- there no need to cram everything into one API.